src/supercontrast/metrics/metrics_factory.py

```diff
@@ -0,0 +1,23 @@
+from typing import Optional
+
+from supercontrast.metrics.metrics_handler import (
+    MetricsHandler,
+    OCRMetricsHandler,
+    TranscriptionMetricsHandler,
+    TranslationMetricsHandler,
+)
+from supercontrast.task.task_enum import Task
+
+
+# metrics factory
+def metrics_factory(task: Task) -> Optional[MetricsHandler]:
+    if task == Task.OCR:
+        return OCRMetricsHandler()
+    elif task == Task.TRANSCRIPTION:
+        return TranscriptionMetricsHandler()
+    elif task == Task.TRANSLATION:
+        return TranslationMetricsHandler()
+    elif task == Task.SENTIMENT_ANALYSIS:
+        return None
+    else:
+        raise ValueError(f"Unsupported task: {task}")
```

Comment:
maybe clarify this error message so its explicit that there is no metric for the given task

Reply:
Actually if there's no metric for the given task we return none, and all tasks should be present in this if block, so we should never reach here unless we haven't covered all tasks which is a logical error that should never happen.